### ISSUE TITLE
feat: HDBSCAN/DBSCAN clustering with elastic distances

### ISSUE THREAD

**Issue by thomaschi78** [drumtorben/polars-ts#91]
## Context

polars-ts currently only supports partition-based clustering (k-medoids, k-Shape) which requires specifying `k` upfront. Density-based methods like HDBSCAN and DBSCAN discover clusters of varying densities without a fixed `k`, making them essential for exploratory analysis.

**Tier: 1 (low-hanging fruit)** — polars-ts already computes pairwise distance matrices in Rust. This just wraps sklearn's HDBSCAN/DBSCAN with `metric="precomputed"`.

**References:**
- McInnes et al. (2017). *hdbscan: Hierarchical density based clustering*. JOSS.
- scikit-learn 1.8+ includes HDBSCAN natively.
- aeon toolkit provides this for time series via elastic distances.

## Proposed API

```python
from polars_ts import hdbscan_cluster, dbscan_cluster

# HDBSCAN — no need to specify k
labels = hdbscan_cluster(df, method="dtw", min_cluster_size=5, min_samples=3)
# Returns: pl.DataFrame with [unique_id, cluster] (noise points get cluster=-1)

# DBSCAN — classic density-based
labels = dbscan_cluster(df, method="dtw", eps=1.5, min_samples=3)
# Returns: pl.DataFrame with [unique_id, cluster] (noise=-1)
```

### Task 1: Add `hdbscan_cluster` function
- **File:** `polars_ts/clustering/density.py` (new)
- Compute pairwise distances via existing `compute_distances()`
- Convert to condensed distance matrix (scipy `squareform`)
- Pass to `sklearn.cluster.HDBSCAN(metric="precomputed")`
- Return `pl.DataFrame` with `[id_col, "cluster"]` schema
- **Acceptance criteria:**
  - [ ] Returns DataFrame with correct schema
  - [ ] Noise points labeled as -1
  - [ ] Supports all existing distance methods (dtw, sbd, msm, etc.)
  - [ ] `min_cluster_size` and `min_samples` parameters exposed
- **Scope:** Small (1 new file)

### Task 2: Add `dbscan_cluster` function
- Same file, same pattern with `sklearn.cluster.DBSCAN(metric="precomputed")`
- Parameters: `eps`, `min_samples`
- **Acceptance criteria:**
  - [ ] Returns DataFrame with correct schema
  - [ ] `eps` and `min_samples` control cluster granularity
  - [ ] Known well-separated data produces correct clusters
- **Scope:** XS (add to existing file)

### Task 3: Register in `__init__.py` and add tests
- Add to `__getattr__`, `__all__`
- Tests: well-separated synthetic data, noise detection, all-noise edge case, custom distance metrics
- **Scope:** Small

## Verification
```bash
uv run pytest tests/clustering/test_density.py -v
```
